great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -458,7 +487,9 @@ def schema_(self) -> str | None:
     @property
     def database(self) -> str | None:
         """Convenience property to get the `database` regardless of the connection string format."""
-        if isinstance(self.connection_string, (ConnectionDetails, SnowflakeDsn)):
+        if isinstance(
```

Comment:
seems like it might be time to abstract this check into its own method so we don't have to update a million isinstance checks. (non-blocking)